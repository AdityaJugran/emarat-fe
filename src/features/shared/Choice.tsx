import { Radio } from 'antd';
import type { RadioChangeEvent } from 'antd';
import type { Dispatch, SetStateAction } from 'react';

type ChoiceProps = {
	choice: number;
	setChoice: Dispatch<SetStateAction<number>>;
	labels: [first: string, second: string];
};

const Choice = ({ setChoice, choice, labels }: ChoiceProps) => {
	const handleChange = (event: RadioChangeEvent) => {
		const { value } = event.target;
		setChoice(value);
	};
	return (
		<Radio.Group
			onChange={handleChange}
			defaultValue={choice}
			buttonStyle="solid"
		>
			<Radio.Button value={0}>{labels[0]}</Radio.Button>
			<Radio.Button value={1}>{labels[1]}</Radio.Button>
		</Radio.Group>
	);
};

export default Choice;
