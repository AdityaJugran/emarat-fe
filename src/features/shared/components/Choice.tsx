import { useHistory } from 'react-router-dom';
import { Radio } from 'antd';
import type { RadioChangeEvent } from 'antd';
import type { Dispatch, SetStateAction } from 'react';

type ChoiceProps = {
	choice: number;
	setChoice: Dispatch<SetStateAction<number>>;
	labels: [first: string, second: string];
	resetUrlOnSetChoice?: {
		on: number;
		url: string;
	};
};

const Choice = (props: ChoiceProps) => {
	const { setChoice, choice, labels, resetUrlOnSetChoice } = props;
	const history = useHistory();

	const handleChange = (event: RadioChangeEvent) => {
		const { value } = event.target;
		setChoice(value);

		if (resetUrlOnSetChoice) {
			const { on, url } = resetUrlOnSetChoice;
			if (on === value) {
				history.push(url);
			}
		}
	};

	return (
		<Radio.Group
			onChange={handleChange}
			value={choice}
			buttonStyle="solid"
			className="title-font"
		>
			<Radio.Button value={0}>{labels[0]}</Radio.Button>
			<Radio.Button value={1}>{labels[1]}</Radio.Button>
		</Radio.Group>
	);
};

export default Choice;
