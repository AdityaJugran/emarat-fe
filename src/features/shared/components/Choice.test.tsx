import { render, fireEvent, screen } from '@testing-library/react';
import Choice from 'features/shared/components/Choice';
import { useState } from 'react';
import renderer from 'react-test-renderer';

const TestChoice = () => {
	const [choice, setChoice] = useState(0);
	return <Choice {...{ choice, setChoice, labels: ['zero', 'one'] }} />;
};

describe('Choice component ', () => {
	it('both labels are in the dom', () => {
		const { getByText } = render(
			<Choice choice={0} labels={['zero', 'one']} setChoice={jest.fn} />
		);
		expect(getByText('zero')).toBeInTheDocument();
		expect(getByText('one')).toBeInTheDocument();
	});

	it('on clicking choice one, snapshot should change', () => {
		render(<TestChoice />);
		const initSnap = renderer.create(<TestChoice />).toJSON();
		fireEvent.click(screen.getByText(/one/i));
		const afterClickSnap = renderer.create(<TestChoice />).toJSON();
		expect(initSnap).not.toBe(afterClickSnap);
	});

	it('Choice snapshot', () => {
		const tree = renderer
			.create(
				<Choice
					choice={0}
					labels={['zero', 'one']}
					setChoice={jest.fn}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
