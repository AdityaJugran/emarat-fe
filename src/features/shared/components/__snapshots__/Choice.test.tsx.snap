// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choice component  Choice snapshot 1`] = `
<div
  className="ant-radio-group ant-radio-group-solid title-font"
>
  <label
    className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
  >
    <span
      className="ant-radio-button ant-radio-button-checked"
      style={Object {}}
    >
      <input
        checked={true}
        className="ant-radio-button-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="radio"
        value={0}
      />
      <span
        className="ant-radio-button-inner"
      />
    </span>
    <span>
      zero
    </span>
  </label>
  <label
    className="ant-radio-button-wrapper"
  >
    <span
      className="ant-radio-button"
      style={Object {}}
    >
      <input
        checked={false}
        className="ant-radio-button-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        type="radio"
        value={1}
      />
      <span
        className="ant-radio-button-inner"
      />
    </span>
    <span>
      one
    </span>
  </label>
</div>
`;
